server/controllers/editFood.js

```diff
@@ -0,0 +1,33 @@
+const Boom = require('@hapi/boom');
+const { editFoodGrams } = require('../database/queries');
+
+const editFood = async (req, res, next) => {
+  const {
+    user: { id },
+  } = req;
+  const { categoryId, foodId } = req.params;
+  const { quantityInGrams } = req.body;
+  try {
+    if (!id) {
+      throw Boom.unauthorized('you are not logged!');
+    }
```

Comment:
I think you could check for negative quantityInGrams values here (or non-numeric ones too) and throw a 400 if there are any problems found.

Reply:
like this?
```
if (quantityInGrams < 0 || typeof quantityInGrams !== 'number') {
      throw Boom.badRequest('invalid amount of grams');
    }
```

Reply:
Yep! If you want, you could move the typeof check first, and the greater than zero second - I’m not sure what javascript will do if you try to check whether a non-number is greater than 0 😅

Reply:
you can just use validate schema from joi that we use
to check validation to quantityInGrams values

Reply:
you can just use validate schema from joi that we use to check validation to quantityInGrams values